uc.c

```diff
@@ -1036,6 +1036,16 @@ uc_err uc_hook_add(uc_engine *uc, uc_hook *hh, int type, void *callback,
         hook->insn = va_arg(valist, int);
         va_end(valist);
 
+
+#ifdef UNICORN_HAS_X86
```

Comment:
there are also few other code in the uc.c that should be isolated in their architectures. while you are at it, if you can fix them all, that will be great!